presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -439,7 +439,7 @@ public static long roundSmallint(@SqlType(StandardTypes.SMALLINT) long num, @Sql
     @Description("round to nearest integer")
     @ScalarFunction("round")
     @SqlType(StandardTypes.INTEGER)
-    public static long roundInteger(@SqlType(StandardTypes.INTEGER) long num, @SqlType(StandardTypes.INTEGER) long decimals)
+    public static long roundInteger(@SqlType(StandardTypes.INTEGER) long num, @SqlType(StandardTypes.BIGINT) long decimals)
```

Comment:
This was actually by design. If you happen to want to round (int, int):int, the function will be resolved to round(int, bigint):int. However, with your design, if you wanted to round(int, bigint):int, the only matching function would be round(bigint, bigint):bigint, which gives you an unexpected result type.